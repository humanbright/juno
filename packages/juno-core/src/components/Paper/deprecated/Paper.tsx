import React, { FunctionComponent } from 'react';

import MuiPaper, { PaperProps as MuiPaperProps } from '@material-ui/core/Paper';

import { RcBaseProps } from '../../../foundation/typings';

export type RcPaperProps = RcBaseProps<MuiPaperProps>;

/** @deprecated please use release Paper */
export const RcPaper: FunctionComponent<RcPaperProps> & {
  dependencies?: any[];
} = React.memo((props) => <MuiPaper {...props} />);

RcPaper.displayName = 'RcPaper';
