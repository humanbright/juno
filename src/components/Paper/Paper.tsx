import MuiPaper from '@material-ui/core/Paper';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineClasses,
  RcBaseProps,
  styled,
  useThemeProps,
} from '../../foundation';
import { PaperStyle } from './styles';
import { RcPaperClasses } from './utils';

type RcPaperProps = {} & RcBaseProps<ComponentProps<typeof MuiPaper>>;

const _RcPaper = forwardRef<any, RcPaperProps>((inProps, ref) => {
  const props = useThemeProps({ props: inProps, name: 'RcPaper' });
  const { classes: classesProp, children, ...rest } = props;
  const classes = useMemo(
    () => combineClasses(RcPaperClasses, classesProp),
    [classesProp],
  );

  return (
    <MuiPaper {...rest} ref={ref} classes={classes}>
      {children}
    </MuiPaper>
  );
});

const RcPaper = styled(_RcPaper)`
  ${PaperStyle}
`;

RcPaper.defaultProps = {};

RcPaper.displayName = 'RcPaper';

export { RcPaper };
export type { RcPaperProps };
