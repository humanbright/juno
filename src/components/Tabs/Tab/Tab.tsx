import MuiTab from '@material-ui/core/Tab';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineProps,
  RcBaseProps,
  RcBaseSize,
  styled,
  useThemeProps,
} from '../../../foundation';
import { TabStyle } from './styles';
import { RcTabClasses } from './utils';

type RcTabSize = RcBaseSize<'small' | 'large'>;

type RcTabProps = {
  /** tab size, default is `small` */
  size?: RcTabSize;
  /** tab wrapper flex direction, default is `vertical` */
  direction?: 'vertical' | 'horizontal';
} & RcBaseProps<ComponentProps<typeof MuiTab>, 'wrapped' | 'textColor'>;

const _RcTab = forwardRef<any, RcTabProps>((inProps, ref) => {
  const props = useThemeProps({ props: inProps, name: 'RcTab' });
  const { classes: classesProp, children, direction, ...rest } = props;
  const classes = useMemo(() => combineProps(RcTabClasses, classesProp), [
    classesProp,
  ]);

  return <MuiTab {...rest} ref={ref} classes={classes} />;
});

const RcTab = styled(_RcTab)`
  ${TabStyle}
`;

RcTab.defaultProps = {
  size: 'small',
};

RcTab.displayName = 'RcTab';

export { RcTab, RcTabProps };
