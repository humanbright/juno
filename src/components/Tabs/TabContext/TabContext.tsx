import MuiTabContext, {
  getPanelId,
  getTabId,
  useTabContext,
} from '@material-ui/lab/TabContext';
import { ComponentProps, FunctionComponent } from 'react';

const RcTabContext = MuiTabContext as FunctionComponent<
  ComponentProps<typeof MuiTabContext>
>;

RcTabContext.displayName = 'RcTabContext';

export { RcTabContext, useTabContext, getPanelId, getTabId };
