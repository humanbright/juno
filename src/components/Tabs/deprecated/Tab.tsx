import { Component, ReactNode } from 'react';

type RcTabProps = {
  /** Tab title */
  title: JSX.Element | string;
  /** Tab content */
  children: ReactNode; // Container
  /** automationId, used for test */
  automationId?: string;
};

/** @release */
class RcTab extends Component<RcTabProps> {
  constructor(props: RcTabProps) {
    super(props);
  }

  render() {
    const { children } = this.props;
    return { children };
  }
}

export { RcTab, RcTabProps };
