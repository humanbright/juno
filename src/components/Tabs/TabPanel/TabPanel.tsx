import MuiTabPanel from '@material-ui/lab/TabPanel';
import React, { ComponentProps, forwardRef, useMemo } from 'react';

import {
  combineClasses,
  RcBaseProps,
  styled,
  useThemeProps,
} from '../../../foundation';
import { useTabContext } from '../TabContext';
import { TabPanelStyle } from './styles';
import { RcTabPanelClasses } from './utils';

type RcTabPanelProps = {} & RcBaseProps<ComponentProps<typeof MuiTabPanel>>;

const _RcTabPanel = forwardRef<any, RcTabPanelProps>(
  (inProps: RcTabPanelProps, ref) => {
    const props = useThemeProps({ props: inProps, name: 'RcTabPanel' });
    const { classes: classesProp, children, ...rest } = props;
    const classes = useMemo(
      () => combineClasses(RcTabPanelClasses, classesProp),
      [classesProp],
    );

    // * that just check is that have context, others let mui handle inner
    const context = useTabContext();
    if (context === null) {
      throw new TypeError('[RcTabList] No RcTabContext provided');
    }

    return (
      <MuiTabPanel {...rest} ref={ref} classes={classes}>
        {children}
      </MuiTabPanel>
    );
  },
);

const RcTabPanel = styled(_RcTabPanel)`
  ${TabPanelStyle}
`;

RcTabPanel.defaultProps = {};

RcTabPanel.displayName = 'RcTabPanel';

export { RcTabPanel, RcTabPanelProps };
