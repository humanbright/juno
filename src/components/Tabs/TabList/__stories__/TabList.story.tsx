import { Meta, Story } from '@storybook/react/types-6-0';
import React, { ComponentProps } from 'react';

import {
  notControlInDocTable,
  notShowInDocTable,
  sortInDocTable,
  switchToControlKnobs,
} from '../../../../storybook';
import { RcPaper } from '../../../Paper';
import { RcTab } from '../../Tab';
import { RcTabContext } from '../../TabContext';
import { RcTabPanel } from '../../TabPanel';
import { RcTabList } from '../TabList';

export default {
  title: 'ðŸš€ Cleanup Components/Tabs/TabContext/TabList',
  component: RcTabList,
  argTypes: {
    ...sortInDocTable<keyof TabListProps>([
      'variant',
      'size',
      'orientation',
      'centered',
      'scrollButtons',
      'selectionFollowsFocus',
    ]),
    ...notControlInDocTable<keyof TabListProps>([]),
    ...notShowInDocTable<keyof TabListProps>([]),
  },
} as Meta;

const tabsData = [
  { label: 'Tab 0', value: 'tab-0', content: 'Tab Content 0' },
  { label: 'Tab 1', value: 'tab-1', content: 'Tab Content 1' },
  { label: 'Tab 2', value: 'tab-2', content: 'Tab Content 2' },
  { label: 'Tab 3', value: 'tab-3', content: 'Tab Content 3' },
  { label: 'Tab 4', value: 'tab-4', content: 'Tab Content 4' },
  {
    label: 'Tab 5',
    value: 'tab-5',
    disabled: true,
    content: 'Tab Content 5',
  },
  { label: 'Tab 6', value: 'tab-6', content: 'Tab Content 6' },
  {
    label: 'Tab 77777777',
    value: 'tab-7',
    content: 'Tab Content 7',
  },
  { label: 'Tab 8', value: 'tab-8', content: 'Tab Content 8' },
  { label: 'Tab 9', value: 'tab-9', content: 'Tab Content 9' },
];

type TabListProps = ComponentProps<typeof RcTabList>;

export const TabList: Story<TabListProps> = ({ ...args }) => {
  switchToControlKnobs();

  const [value, setValue] = React.useState(tabsData[0].value);

  const handleChange = (event: React.ChangeEvent<{}>, value: any) => {
    setValue(value);
  };

  const TabChildren = tabsData.map((tab) => {
    const { label, value, disabled, content, ...rest } = tab;
    return (
      <RcTab
        key={label}
        label={label}
        value={value}
        disabled={disabled}
        {...rest}
      />
    );
  });

  const TabContentChildren = tabsData.map((tab) => {
    const { value, content } = tab;
    return (
      <RcTabPanel key={value} value={value}>
        {content}
      </RcTabPanel>
    );
  });

  return (
    <RcTabContext value={value}>
      <RcPaper square>
        <RcTabList
          {...args}
          onChange={handleChange}
          aria-label="simple tabs example"
        >
          {TabChildren}
        </RcTabList>
      </RcPaper>
      {TabContentChildren}
    </RcTabContext>
  );
};

TabList.storyName = 'TabList';

TabList.args = {};

TabList.argTypes = {
  ...notControlInDocTable<keyof TabListProps>([]),
};
