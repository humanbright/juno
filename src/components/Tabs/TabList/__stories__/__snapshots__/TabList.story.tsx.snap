// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots ðŸš€ Cleanup Components/Tabs/TabContext/TabList TabList 1`] = `
Array [
  .c1 {
  text-transform: none;
  box-sizing: border-box;
  font-size: 0.9375rem;
  font-weight: 400;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #121212;
  min-height: 32px;
  min-width: 32px;
}

.c1.RcTab-selected {
  font-size: 0.9375rem;
  font-weight: 700;
  font-family: Lato,Helvetica,Arial,sans-serif;
  line-height: 22px;
  color: #066FAC;
}

.c0 {
  min-height: auto;
}

.c0 .RcTabs-indicator {
  background-color: #066FAC;
}

<div
    className="MuiPaper-root MuiPaper-elevation1"
  >
    <div
      className="MuiTabs-root c0 "
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": null,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-label="simple tabs example"
          className="MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-controls={null}
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1 Mui-selected RcTab-selected"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 0
            </span>
            <span
              className="PrivateTabIndicator-root-187 PrivateTabIndicator-colorPrimary-188 MuiTabs-indicator RcTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 1
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 2
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 3
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 4
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1 Mui-disabled Mui-disabled"
            disabled={true}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 5
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 6
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 77777777
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 8
            </span>
          </button>
          <button
            aria-controls={null}
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary c1"
            disabled={false}
            id={null}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            size="small"
            tabIndex={-1}
            type="button"
          >
            <span
              className="MuiTab-wrapper RcTab-wrapper"
            >
              Tab 9
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={false}
    id={null}
    role="tabpanel"
  >
    Tab Content 0
  </div>,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
  <div
    aria-labelledby={null}
    className="MuiTabPanel-root "
    hidden={true}
    id={null}
    role="tabpanel"
  />,
]
`;
